src/graph/LookupExecutor.cpp

```diff
@@ -126,15 +130,23 @@ Status LookupExecutor::prepareYield() {
                 return Status::SyntaxError("Not supported yet");
             }
             auto* prop = col->expr();
-            if (prop->kind() != Expression::kAliasProp) {
-                return Status::SyntaxError("Expressions other than AliasProp are not supported");
+            const AliasPropertyExpression* aExpr;
+            if (prop->kind() == Expression::kAliasProp) {
+                aExpr = dynamic_cast<const AliasPropertyExpression*>(prop);
+            } else if (isEdge_ && prop->kind() == Expression::kEdgeSrcId) {
+                aExpr = dynamic_cast<const EdgeSrcIdExpression*>(prop);
+            } else if (isEdge_ && prop->kind() == Expression::kEdgeDstId) {
+                aExpr = dynamic_cast<const EdgeDstIdExpression*>(prop);
+            } else {
+                return Status::SyntaxError("Expressions other than "
+                                           "AliasProp/kEdgeSrcId/kEdgeDstId are not supported");
```

Comment:
Is there a problem with the `dynamic_cast` here?

Reply:
Thanks for your review! EdgeSrcIdExpression and EdgeDstIdExpression both extend AliasPropertyExpression, and I tested these logical branches, no problem found here.

Reply:
> Thanks for your review! EdgeSrcIdExpression and EdgeDstIdExpression both extend AliasPropertyExpression, and I tested these logical branches, no problem found here.

Got , if so , I think below code is better

```
if (prop->kind() == Expression::kAliasProp) {
                 aExpr = static_cast<const AliasPropertyExpression*>(prop);
             } else if (isEdge_ && prop->kind() == Expression::kEdgeSrcId) {
                 aExpr = static_cast <const AliasPropertyExpression*>(prop);
             } else if (isEdge_ && prop->kind() == Expression::kEdgeDstId) {
                 aExpr = static_cast <const AliasPropertyExpression*>(prop);
             } else {
                      return ERROR;
             }
```

OR 

```
if (prop->kind() != Expression::kAliasProp && isEdge_) {
   if (prop->kind() != Expression::kEdgeSrcId || prop->kind() != Expression::kEdgeDstId) {
                      return ERROR;
  }
}
auto* aExpr = dynamic_cast<const AliasPropertyExpression*>(prop);
```

Reply:
All right. I have completed the modification as suggested, please check it.